enhancements/declarative-index-config.md

```diff
@@ -0,0 +1,947 @@
+
+## Non-goals 
+
+1. Enumerate/implement tooling to allow different operations to be performed on the package representations. Numerous tooling(eg those that allow operation on files) that already exists can be leveraged to perform various operations on package representations. 
+2. Advanced tooling to verify dependency satisfiability for individual bundles, channel upgrade graph validity for each channel in a package etc. This will be considered in a separate enhancement.      
+
+## User Stories 
+
+### Story 1
+
+As a user with pull permission from the namespace an index is hosted in/as a component of OLM, I can query an index for a json/yaml representation of the packages of an index.
+The representation of individual packages will allow individual package owners to reason about/make changes to their individual packages in isolation.  
+
+`opm index inspect --image=docker.io/my-namespace/community-operators --output=json`
+
+pulls the index image from the registry, creates a new folder `community-operators` and copies the package representations for the packages that are in the image to the local folder. 
+
+```bash
+$ cd community-operators
+$ tree
+.
+├── etcd.json
+├── amqstreams.json
+
+$ cat etcd.json
+{
+    "schema": "olm.package",
+    "name": "etcd",
+    "defaultChannel": "singlenamespace-alpha",
+    "icon": {
+        "base64data":"iVBORw0KGgoAAAANSUhEUgAAA.....",
+        "mediatype":"image/png"
+    },
+    "channels": ["alpha", "singlenamespace-alpha", "clusterwide-alpha"],
+    "description": "A message about etcd operator, a description of channels"
+},
+{
+    "schema": "olm.bundle", 
+    "name": "etcdoperator-community.v0.6.1",
+    "package": "etcd",
+    "image": "quay.io/operatorhubio/etcd:v0.6.1",
+    "version": "0.6.1",
+    "properties":[
+        {
+            "name": "olm.package",
+            "values": {
+                "packageName": "etcd",
+                "version": "0.6.1"
+            }
+        },
+        {
+            "name":"olm.gvk",
+            "type": "provided",
+            "values":{
+                "group": "etcd.database.coreos.com", 
+                "kind": "EtcdCluster", 
+                "version": "v1beta2"
+            }
+        },
+        {
+            "name": "olm.channel",
+            "value": "alpha"
+        }
+    ],
+    "relatedImages": [
+        {
+            "name": "etcdv0.6.1",
+            "image": "quay.io/coreos/etcd-operator@sha256:bd944a211eaf8f31da5e6d69e8541e7cada8f16a9f7a5a570b22478997819943"
+        }
+    ]
+},
+{
+    "schema": "olm.bundle",
+    "name": "etcdoperator.v0.9.0",
+    "package": "etcd",
+    "image": "quay.io/operatorhubio/etcd:v0.9.0",
+    "version": "0.9.0",
+    "properties":[
+        {
+            "name": "olm.package",
+            "values": {
+                "packageName": "etcd",
+                "version": "0.9.0"
+            }
+        },
```

Comment:
This may have already been discussed, but I'm curious if it would make sense/be possible to make the property type simpler? Trying to model what's described in this EP, we would need:
```go
type Property struct
	Name string
	Values map[string]string // Optional: only used with olm.gvk and olm.package
	Type string              // Optional: only used with olm.gvk
    Replaces string          // Optional: only used with olm.channel 
	Value string             // Optional: only used with olm.channel
}
```

I would propose simplifying to:
```go
type Property struct
	Name string
	Values map[string]string
}
```
* `olm.package` would not change.
* For `olm.gvk`, I would propose moving `type` into the values section or combining name and type (e.g. `olm.gvk.provided` and `olm.gvk.required`).
* For `olm.channel`, I would propose `values: { name: "my-channel", replaces: "my-operator.v0.0.1"}`